addons/web/controllers/action.py

```diff
@@ -45,3 +50,37 @@ def run(self, action_id, context=None):
         action = request.env['ir.actions.server'].browse([action_id])
         result = action.run()
         return clean_action(result, env=action.env) if result else False
+
+    @route('/web/action/load_breadcrumbs', type='json', auth='user', readonly=True)
+    def load_breadcrumbs(self, actions):
+        display_names = []
+        for action in actions:
+            record_id = action.get('resId')
+            try:
+                if action.get('action'):
+                    act = self.load(action.get('action'))
+                    if (act['type'] == 'ir.actions.server'):
+                        if (act['path']):
+                            act = request.env['ir.actions.server'].browse(act['id']).run()
```

Comment:
browsing in a loop is usually a bad practice, this will break batch prefetch from the orm.

I don't think there will be many actions, but in any case it could be interesting to improve that slighly by listing all 'ir.actions.server' ids, browsing and iteration on this record set.

This is not as easy to do for the second browse since all models may be unique.

Reply:
This is a corner-case, it's exclusively for crm and to-do, that they use server's actions on the menu. We will, in a future, investigate to change this.